schemaregistry/schemaregistry_client.go

```diff
@@ -260,18 +264,34 @@ func NewClient(conf *Config) (Client, error) {
 		if err != nil {
 			return nil, err
 		}
+		latestToSchemaCache, err = cache.NewLRUCache(conf.CacheCapacity)
+		if err != nil {
+			return nil, err
+		}
 	} else {
 		schemaToIDCache = cache.NewMapCache()
 		idToSchemaCache = cache.NewMapCache()
 		schemaToVersionCache = cache.NewMapCache()
 		versionToSchemaCache = cache.NewMapCache()
+		latestToSchemaCache = cache.NewMapCache()
+	}
+	if conf.CacheLatestTTLSecs > 0 {
+		go func() {
+			ticker := time.NewTicker(time.Duration(conf.CacheLatestTTLSecs) * time.Second)
+			defer ticker.Stop()
+			for {
+				<-ticker.C
+				latestToSchemaCache.Clear()
+			}
+		}()
 	}
 	handle := &client{
 		restService:          restService,
 		schemaToIDCache:      schemaToIDCache,
 		idToSchemaCache:      idToSchemaCache,
 		schemaToVersionCache: schemaToVersionCache,
 		versionToSchemaCache: versionToSchemaCache,
+		latestToSchemaCache:  latestToSchemaCache,
 	}
 	return handle, nil
```

Comment:
Coulse use `runtime.SetFinalizer` with `handle` that stops the ticker and a done channel to exit the for loop in the spawned goroutine

Reply:
Yes, let me make that change

Reply:
@emasab , I added the finalizer